src/sentry/processing/issues_by_tag.py

```diff
@@ -0,0 +1,102 @@
+
+import six
+
+from sentry import tagstore
+from sentry.models import EventUser, Group, get_group_with_redirect
+from .base import BaseProcessor, ProcessingError
+
+
+class IssuesByTagProcessor(BaseProcessor):
+    def __init__(self, project_id, group_id, key, environment_id):
+        self.project = self.get_project(project_id)
+        self.group = self.get_group(group_id, self.project)
+        self.key = key
+        self.environment_id = environment_id
+        self.fields = self.get_fields(self.key)
+        self.lookup_key = self.get_lookup_key(self.key)
+        # Ensure the tag key exists, as it may have been deleted
+        try:
+            tagstore.get_tag_key(self.project.id, environment_id, self.lookup_key)
+        except tagstore.TagKeyNotFound:
+            raise ProcessingError("Requested key does not exist")
+        self.callbacks = self.get_callbacks(self.key, self.group.project_id)
+
+    @staticmethod
+    def get_group(group_id, project):
+        try:
+            group, _ = get_group_with_redirect(
+                group_id, queryset=Group.objects.filter(project=project)
+            )
+            return group
+        except Group.DoesNotExist:
+            raise ProcessingError("Requested issue does not exist")
+
+    @staticmethod
+    def get_fields(key):
+        if key == "user":
+            return [
+                "value",
+                "id",
+                "email",
+                "username",
+                "ip_address",
+                "times_seen",
+                "last_seen",
+                "first_seen",
+            ]
+        else:
+            return ["value", "times_seen", "last_seen", "first_seen"]
+
+    @staticmethod
+    def get_lookup_key(key):
+        return six.text_type("sentry:{}".format(key)) if tagstore.is_reserved_key(key) else key
+
+    @staticmethod
+    def get_eventuser_callback(project_id):
+        def attach_eventuser(items):
+            users = EventUser.for_tags(project_id, [i.value for i in items])
+            for item in items:
+                item._eventuser = users.get(item.value)
+
+        return attach_eventuser
+
+    @staticmethod
+    def get_callbacks(key, project_id):
+        return [IssuesByTagProcessor.get_eventuser_callback(project_id)] if key == "user" else []
+
+    @staticmethod
+    def serialize_issue(item, key):
+        result = {
+            "value": item.value,
+            "times_seen": item.times_seen,
+            "last_seen": item.last_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+            "first_seen": item.first_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+        }
+        if key == "user":
+            euser = item._eventuser
+            result["id"] = euser.ident if euser else ""
+            result["email"] = euser.email if euser else ""
+            result["username"] = euser.username if euser else ""
+            result["ip_address"] = euser.ip_address if euser else ""
+        return result
+
```

Comment:
The methods above this comment are all `@staticmethod`s, and can be reused if need be (see `group_tag_export`'s `get_header` and `get_row` methods for an example).

The methods below are not static, and require an instance, meaning the `__init__` constructor has to pass all the way through. In passing, it runs most of the above functions, and sets up the two below to actually do the work (i.e. create lists of GroupTagValues)